common/v2/components/TransactionFlow/TxReceipt.tsx

```diff
@@ -76,34 +82,68 @@ export default function TxReceipt({
   const { assets } = useContext(AssetContext);
   const { networks } = useContext(NetworkContext);
   const [txStatus, setTxStatus] = useState(ITxStatus.PENDING);
-  const [displayTxReceipt, setDisplayTxReceipt] = useState(txReceipt as ITxReceipt);
+  const [displayTxReceipt, setDisplayTxReceipt] = useState<ITxReceipt | undefined>(txReceipt);
   const [blockNumber, setBlockNumber] = useState(0);
   const [timestamp, setTimestamp] = useState(0);
 
+  const protectTxContext = useContext(ProtectTxContext);
+  const getProTxValue = ProtectTxUtils.isProtectTxDefined(protectTxContext);
+
   useEffect(() => {
-    const provider = new ProviderHandler(displayTxReceipt.network || txConfig.network);
-    if (blockNumber === 0 && displayTxReceipt.hash) {
+    setDisplayTxReceipt(txReceipt);
+  }, [setDisplayTxReceipt, txReceipt]);
+
+  const [protectTxCounter, setProtectTxCounter] = React.useState(20);
+  useEffect(() => {
+    let protectTxTimer: ReturnType<typeof setTimeout> | null = null;
+    if (
+      !getProTxValue(['state', 'isWeb3Wallet']) &&
+      getProTxValue(['state', 'protectTxEnabled']) &&
```

Comment:
This syntax is surprising.
Particularly on L111 with `getProTxValue(['invokeProtectTxTimeoutFunction']`.
Do we really need it? It seems to be just as verbose as:
```
const { state, invokeProtectTxTimeoutFunction } = useContext(ProtectTxContext)

.../
if (state && state.isWeb3Wallet && state.protectTextEnabled && ...)
```

and since you call it multiple time you can de-structure state as well:
```
const { isWeb3Wallet, protectTxEnabled } = state
```

At which point the variables should be reactive as well. And of course there is also R.prop, R.path and R.hasPaths to use and abuse.

Reply:
Well, the problem here is that the `ProtectTxContext` may not be always available in `TxReciept`, since it is only used in send assets flow. So destructuring won't work. So I wanted to take the similar approach as R.path, and not always pass in the context, to make it more readable. The big downside is that type checks are not available, when accessing properties by strings. I think R approach would be less readable, in my opinion.

Reply:
Very well. In any case it should be easier once it's moved into the Provider. Ping me on slack and I'll merge.